### ISSUE TITLE
False `check-extra-masked-returns` positive on assignment and multiple command substitutions

### ISSUE THREAD

**Issue by scop** [koalaman/shellcheck#2392]
#### For bugs
- Rule Id (if any, e.g. SC2312): 
- My shellcheck version (`shellcheck --version` or "online"): 0.8.0
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2312)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [ ] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
# shellcheck enable=check-extra-masked-returns
x="$(foo)/$(foo)"
```

#### Here's what shellcheck currently says:

```
     ^-- SC2312 (info): Consider invoking this command separately to avoid masking its return value (or use '|| true' to ignore).
```

#### Here's what I wanted or expected to see:

No SC2312. It appears that the second command substitution provokes this false positive, with just `x="$(foo)"` it is not emitted (as it should not be, as this is an assignment that does get terminated in `-e` mode, not an argument to a command which the Wiki page mentions).

**Comment by scop** [comment]
Hmm, seems this is not a false positive after all, the second command substitution does indeed mask the return value of the first (assignment doesn't error out in `-e` mode if the latter substitution succeeds), e.g. `x="$(this_will_fail)/$(true)"`

Would be good to make the wiki page cover this case as well as other possible ones and not talk just about command substitution in argument of another command, leaving open for someone who knows the specifics to address.
